evap/staff/importers.py

```diff
@@ -417,6 +470,93 @@ def process(cls, excel_content, test_run):
                 raise
 
 
+class PersonImporter:
+    def __init__(self, import_type, course):
+        self.import_type = import_type
+        self.course = course
+        self.user_list = []
+        self.users_to_add = []
+
+        self.success_messages = []
+        self.warnings = defaultdict(list)
+        self.errors = []
+
+    def handle_already_related(self):
+        already_related = []
+
+        if self.import_type == "participant":
+            usernames = [user.username for user in self.user_list]
+            already_related = self.course.participants.filter(username__in=usernames).all()
+        else:  # import_type == "contributor"
+            already_related = Contribution.objects.filter(course=self.course, contributor__in=self.user_list).all()
```

Comment:
If I understand you correctly, then this code should fail (since a newly created UserProfile doesn't have an id, right?):
```
u = [UserProfile()]
already_related = Contribution.objects.filter(course=self.course, contributor__in=(self.user_list + u)).all()
```
It works fine although I can't find any guarantees on that in the documentation. Just appending an integer works too. A string fails, the error message being that it can't be casted to int. What are your thoughts? Is relying on "it's working right now" okay or do we need guarantees in the documentation?